packages/astro/src/@types/astro.ts

```diff
@@ -1874,10 +1876,17 @@ export interface AstroIntegration {
 		'astro:config:done'?: (options: {
 			config: AstroConfig;
 			setAdapter: (adapter: AstroAdapter) => void;
+			logger: AstroIntegrationLogger;
 		}) => void | Promise<void>;
```

Comment:
I wonder if we can/should only expose the logger here. Then integrations can assign this to a variable outside of the hook and use it everywhere, as I think that's commonly needed too when abstracting functions.

Reply:
We would force the consumers to use this "trick" to save the logger instance. I prefer to give our users everything they need in the hooks.

It's a pity we don't have a high-order function to define integrations because we could pass the logger there.

Reply:
I think it's mostly me accustomed to Vite's flow, which is something like: 

```js
let resolvedConfig

return {
  name: 'vite-plugin',
  configResolved(config) {
    resolvedConfig = config
  },
  transform() {
    resolvedConfig.logger.warn('...')
  }
}
```

Not fully oppose if we're ok with exposing `logger` in most of the hooks though.